base_classes/NXbeam.nxdl.xml

```diff
@@ -245,6 +269,91 @@
             <dim index="1" value="nP"/>
         </dimensions>
     </field>
+    <field name="pulse_energy" type="NX_FLOAT" units="NX_ENERGY">
+        <doc>
+             Energy of a single pulse at the diagnostic point
+        </doc>
+    </field>
+    <field name="average_power" type="NX_FLOAT" units="NX_POWER">
+        <doc>
+             Average power at the diagnostic point
+        </doc>
+    </field>
+    <field name="fluence" type="NX_FLOAT" units="mJ/cm^2">
+        <doc>
+             Incident fluence at the diagnostic point
+        </doc>
+    </field>
+    <field name="pulse_duration" type="NX_FLOAT" units="NX_TIME">
+        <doc>
+             FWHM duration of the pulses at the diagnostic point
+        </doc>
+    </field>
+    <field name="pulse_delay" type="NX_FLOAT" units="NX_TIME">
+        <doc>
+             Delay time between two pulses of a pulsed beam.
+        </doc>
+        <attribute name="reference_beam" type="NX_CHAR">
+            <doc>
+                 A reference to the beam in relation to which the delay is.
+            </doc>
+        </attribute>
+    </field>
+    <field name="frog_trace" type="NX_FLOAT">
+        <doc>
+             FROG (frequency-resolved optical gating) trace of the pulse.
+
+             This is to be used for ultrashort laser pulses in a
+             FROG (frequency-resolved optical gating) setup.
+        </doc>
+        <dimensions rank="2">
+            <dim index="1" value="nx"/>
+            <dim index="2" value="ny"/>
+        </dimensions>
+    </field>
+    <field name="frog_delays" type="NX_FLOAT" units="NX_TIME">
+        <doc>
+             Horizontal axis of a FROG trace, i.e. delay.
+
+             This is to be used for ultrashort laser pulses in a
+             FROG (frequency-resolved optical gating) setup.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="nx"/>
+        </dimensions>
+    </field>
+    <field name="frog_frequencies" type="NX_FLOAT" units="NX_FREQUENCY">
+        <doc>
+             Vertical axis of a FROG trace, i.e. frequency.
+
+             This is to be used for ultrashort laser pulses in a
+             FROG (frequency-resolved optical gating) setup.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="ny"/>
+        </dimensions>
+    </field>
+    <field name="chirp_type" type="NX_CHAR">
+        <doc>
+             The type of chirp implemented
+        </doc>
+    </field>
+    <field name="chirp_GDD" type="NX_FLOAT" units="NX_TIME" nameType="specified">
+        <doc>
+             Group delay dispersion of the pulse for linear chirp
+        </doc>
+    </field>
+    <field name="previous_device">
+        <doc>
+             Indicates the beam device from which this beam originates.
+             This defines whether the beam is an "input" or "output" beam.
+        </doc>
+    </field>
+    <field name="next_device">
```

Comment:
I'm assuming previous_device and next_device are NX_CHAR with the full path to group? We probably want to clarify this.

Reply:
Similar to https://manual.nexusformat.org/classes/base_classes/NXtransformations.html#nxtransformations-axisname-depends-on-attribute I presume.

Reply:
Resolved in the Telco